ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/task/AbstractRetryingPeerTask.java

```diff
@@ -93,11 +94,21 @@ protected void executeTask() {
               if (error != null) {
                 handleTaskError(error);
               } else {
-                // If we get a partial success, reset the retry counter.
-                if (!isEmptyResponse.test(peerResult)) {
-                  retryCount = 0;
+                if (successfulResult(peerResult)) {
+                  result.complete(peerResult);
```

Comment:
I wonder how it worked before without this success part. Is this something that is done automatically and that we don't really need?

Reply:
Before each extending task had to set the final result by itself, using `result.complete`, and another thing that was a bit confusing to me, is that the returned value of a task is not the final result until you set `result.complete` and so I tried to centralize this part here and make the writing of extending tasks easier.

Reply:
good idea